pandas/core/ops.py

```diff
@@ -594,20 +594,26 @@ def wrapper(self, other):
 
         mask = isnull(self)
 
-        values = self.get_values()
-        other = _index.convert_scalar(values,_values_from_object(other))
+        if com.is_categorical_dtype(self):
+            # cats are a special case as get_values() would return an ndarray, which would then
```

Comment:
This feels a bit hacky to me. ideally, shouldn't blocks contain all the logic for arithmetic operations? That way, the special casing for different dtypes would just be a different methods defined on block subclasses, not a mess of dtype checks on the series itself.

This is not your technical debt, though, so we can save that cleanup for later.